### ISSUE TITLE
getCurve causes "undefined is not a function" error

### ISSUE THREAD

**Issue by jakubszlendak** [DevelopmentIL/ecdh#1]
When calling this code:

'''
/media/win/workspace/node/ecdh_key_exchange/node_modules/ecdh/jsbn/ec.js:265
    this.reducer = new Barrett(this.q);
                   ^
TypeError: undefined is not a function
    at new ECCurveFp (/media/win/workspace/node/ecdh_key_exchange/node_modules/ecdh/jsbn/ec.js:265:20)
    at Object.secp128r1 (/media/win/workspace/node/ecdh_key_exchange/node_modules/ecdh/jsbn/sec.js:54:17)
    at Object.exports.getCurve (/media/win/workspace/node/ecdh_key_exchange/node_modules/ecdh/index.js:15:29)
    at Object.<anonymous> (/media/win/workspace/node/ecdh_key_exchange/node_modules/ecdh/examples/ecdh.js:4:18)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
'''

This is present either on my local node installation or in NPM "Try it out" option

**Comment by moshest** [comment]
Works on my side. Looks like `Barrett` is undefined on your code.

**Comment by jakubszlendak** [comment]
Isn't Barret a part of jsbn? Shouldn't there be an error in code below (it's top of ec.js) then?:

var Barrett = BigInteger.prototype.Barrett

Sorry if my questions are stupid, I'm totally new in Node, till now I've written only in C ;)

**Comment by moshest** [comment]
`Barrett` is part of `jsbn` module. It's look like it's has been changed.
Try `npm install ecdh & node ./node_modules/ecdh/examples/ecdh.js` and check if you have any error.

**Comment by jakubszlendak** [comment]
I tried exactly what you suggested but with no effect. May I have wrong node version? I have v0.10.40

**Comment by moshest** [comment]
Looks like there was change on module `jsbn` without a version update (That's shouldn't happened!).
You can try to download the original library from [here](https://github.com/andyperlitch/jsbn) and put it on `node_modules` folder.

Push request with a fix is also welcome.

**Comment by rubyjohn** [comment]
You must fix jsbn's index.js.

----------------------------------------------------------------------------------------
(jsbn version 0.0.0)
Path:
/node_modules/jsbn/index.js
Or
/node_modules/ecdh/node_modules/jsbn/index.js


Code:
<Original Code>
...
    // JSBN-specific extension
    BigInteger.prototype.square = bnSquare;

    // BigInteger interfaces not implemented in jsbn:
...

<New Code>
...
    // JSBN-specific extension
    BigInteger.prototype.square = bnSquare;

    // Expose the Barrett function
    BigInteger.prototype.Barrett = Barrett;

    // BigInteger interfaces not implemented in jsbn:
...


----------------------------------------------------------------------------------------
(jsbn version 0.1.0)
Path:
/node_modules/jsbn/index.js

<Original Code>
...
    if (typeof exports !== 'undefined') {
      exports = module.exports = {
  			BigInteger: BigInteger,
  			SecureRandom: SecureRandom,
  		};
    } else {
        this.BigInteger = BigInteger;
        this.SecureRandom = SecureRandom;
    }
...

<New Code>
 if (typeof exports !== 'undefined') {
	exports = module.exports = BigInteger;
    } else {
        this.BigInteger = BigInteger;
        this.SecureRandom = SecureRandom;
    }
----------------------------------------------------------------------------------------

**Comment by rubyjohn** [comment]
You must fix jsbn's index.js.
v0.0.0 and v0.1.0 are different.

---

##### jsbn version 0.0.0

Path:

/node_modules/jsbn/index.js
Or
/node_modules/ecdh/node_modules/jsbn/index.js

Code:

<Original Code>

``` javascript
...
    // JSBN-specific extension
    BigInteger.prototype.square = bnSquare;

    // BigInteger interfaces not implemented in jsbn:
...
```

## 

<New Code>

``` javascript
...
    // JSBN-specific extension
    BigInteger.prototype.square = bnSquare;

    // Expose the Barrett function
    BigInteger.prototype.Barrett = Barrett;

    // BigInteger interfaces not implemented in jsbn:
...
```

---

##### jsbn version 0.1.0

Path:
/node_modules/jsbn/index.js

Code:

<Original Code>

``` javascript
...
    if (typeof exports !== 'undefined') {
        exports = module.exports = {
            BigInteger: BigInteger,
            SecureRandom: SecureRandom,
        };
    } else {
        this.BigInteger = BigInteger;
        this.SecureRandom = SecureRandom;
    }
...
```

## 

<New Code>

``` javascript
...
    if (typeof exports !== 'undefined') {
        exports = module.exports = BigInteger;
    } else {
        this.BigInteger = BigInteger;
        this.SecureRandom = SecureRandom;
    }
...
```

---

**Comment by moshest** [comment]
fixed on v0.1.1
